app/src/app.rs

```diff
@@ -47,6 +47,103 @@ pub async fn initialize_state_from_genesis(state: &mut State, engine: &Engine) -
     Ok(())
 }
 
+/// Replay blocks from Emerald's store to the execution client (Reth).
+/// This is needed when Reth is behind Emerald's stored height after a crash.
+async fn replay_heights_to_engine(
+    state: &State,
+    engine: &Engine,
+    start_height: Height,
+    end_height: Height,
+    emerald_config: &EmeraldConfig,
+) -> eyre::Result<()> {
+    info!(
+        "🔄 Replaying heights {} to {} to execution client",
+        start_height, end_height
+    );
+
+    for height in start_height.as_u64()..=end_height.as_u64() {
+        let height = Height::new(height);
+
+        // Get the certificate and header from store
+        let (_certificate, header_bytes) = state
+            .store
+            .get_certificate_and_header(height)
+            .await?
+            .ok_or_eyre(format!("Missing certificate or header for height {height}"))?;
+
+        // Deserialize the execution payload
+        let execution_payload = ExecutionPayloadV3::from_ssz_bytes(&header_bytes).map_err(|e| {
+            eyre!(
+                "Failed to deserialize execution payload at height {}: {:?}",
+                height,
+                e
+            )
+        })?;
+
+        debug!(
+            "🔄 Replaying block at height {} with hash {:?}",
+            height, execution_payload.payload_inner.payload_inner.block_hash
+        );
+
+        // Extract versioned hashes from blob transactions
+        let block: Block = execution_payload.clone().try_into_block().map_err(|e| {
+            eyre!(
+                "Failed to convert execution payload to block at height {}: {}",
+                height,
+                e
+            )
+        })?;
+        let versioned_hashes: Vec<BlockHash> =
+            block.body.blob_versioned_hashes_iter().copied().collect();
+
+        // Submit the block to Reth
+        let payload_status = engine
+            .notify_new_block_with_retry(
+                execution_payload.clone(),
+                versioned_hashes,
+                &emerald_config.retry_config,
+            )
+            .await?;
+
+        // Verify the block was accepted
+        match payload_status.status {
+            PayloadStatusEnum::Valid => {
+                debug!("✅ Block at height {} replayed successfully", height);
+            }
+            PayloadStatusEnum::Invalid { validation_error } => {
+                return Err(eyre::eyre!(
+                    "Block replay failed at height {}: {}",
+                    height,
+                    validation_error
+                ));
+            }
+            PayloadStatusEnum::Accepted => {
```

Comment:
This is invalid in Emerald's use case. Accepted is used in a setting where there is no instant finality and there Is a possibility of a fork so here we can treat it the same way as invalid.